integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -71,28 +75,44 @@ def __init__(self, user, enterprise_configuration):
-        )
-        LOGGER.info(
-            'Getting metadata for Enterprise [%s], Catalogs [%s] from Enterprise Catalog Service. Results: [%s]',
-            self.enterprise_customer.name,
-            self.enterprise_configuration.customer_catalogs_to_transmit,
-            json.dumps(content_metadata_items)
-        )
-        for item in content_metadata_items:
-            transformed = self._transform_item(item)
-            LOGGER.debug(
-                'Exporting content metadata item with plugin configuration [%s]: [%s]',
-                self.enterprise_configuration,
-                json.dumps(transformed, indent=4),
-            )
-            content_metadata_item_export = ContentMetadataItemExport(item, transformed)
-            content_metadata_export[content_metadata_item_export.content_id] = content_metadata_item_export
+        force_retrieve_all_catalogs = kwargs.get('force_retrieve_all_catalogs')
+        for enterprise_customer_catalog in enterprise_customer_catalogs:
+            if not force_retrieve_all_catalogs:
+                # Retrieve the time at which the catalog and it's content were last updated
+                content_last_modified, catalog_modified = self._get_enterprise_catalog_last_changed(
+                    enterprise_customer_catalog
+                )
+                # Retrieve whatever modification happened more recently
+                catalog_last_modified = max(
+                    content_last_modified,
+                    catalog_modified
+                )
+                last_successful_transmission = self._get_most_recent_catalog_update_time()
+
+            # If the last successful transmission was since the last time the catalog was updated, there's no need for
+            # an update
+            if force_retrieve_all_catalogs or not last_successful_transmission or \
+                    last_successful_transmission < catalog_last_modified:
+                content_metadata_item = self.enterprise_catalog_api.get_content_metadata(
+                    self.enterprise_customer,
+                    [enterprise_customer_catalog]
+                )
+                for item in content_metadata_item:
+                    transformed = self._transform_item(item)
+                    LOGGER.debug(
+                        'Exporting content metadata item with plugin configuration [%s]: [%s]',
+                        self.enterprise_configuration,
+                        json.dumps(transformed, indent=4),
+                    )
+                    content_last_modified = item.pop('content_last_modified')
+                    content_metadata_item_export = ContentMetadataItemExport(item, transformed, content_last_modified)
+                    content_metadata_export[content_metadata_item_export.content_id] = content_metadata_item_export
         return OrderedDict(sorted(content_metadata_export.items()))
```

Comment:
If I understand this correctly, you have keys representing course keys: `content_metadata_export[content_metadata_item_export.content_id]`
Why are you sorting the data by course key? Do we really need data sorted alphabetically by course key or should this be some other field?

Reply:
Edit: I know it was already like that, but why?

Reply:
Good point- my only thought is that it could have something to do with the customers who's transmission chunk size is larger and they somehow/for some reason expect the data in sorted order. But I can't point to any one example of that...